library/std/build.rs

```diff
@@ -102,10 +102,7 @@ fn main() {
         // Unsupported <https://github.com/llvm/llvm-project/issues/94434>
         ("arm64ec", _) => false,
         // MinGW ABI bugs <https://gcc.gnu.org/bugzilla/show_bug.cgi?id=115054>
-        ("x86_64", "windows") => false,
-        // x86 has ABI bugs that show up with optimizations. This should be partially fixed with
-        // the compiler-builtins update. <https://github.com/rust-lang/rust/issues/123885>
-        ("x86" | "x86_64", _) => false,
+        ("x86", "windows") => false,
```

Comment:
It was intentional since I _thought_ that https://gcc.gnu.org/bugzilla/show_bug.cgi?id=115054 only applied to 64-bit. But I am less certain now, I'll add a x86-32 Windows try job after I figure out the distcheck failure.

Any ideas for that? I don't know why we would be hitting the LLVM assertion only for that job, and I can't reproduce it locally for better debugging.

Reply:
From a quick look https://gcc.godbolt.org/z/Yq5sMGcvr it appears that GCC does the same thing for its x86-32 linux and MinGW targets. I honestly have no clue what clang is doing as far as why it is producing two extra labels per function and `call`ing them. Aside from that, looks like it is doing stack alignment and GCC isn't - but also for the Linux version? Maybe my args are wrong.

In any case, I'll just make this `("x86" | "x86_64)` and address Windows at a later time.

Reply:
> AFAIK the MinGW bug does only apply to 64-bit; I was asking as this PR currently changes the ignored platform from to `("x86_64", "windows")` to `("x86", "windows")`.

Sorry, you're right. I was reading your suggested diff rather than the actual diff. I will still just leave both disabled for now.

> I can reproduce the CI failure locally (on `x86_64-unknown-linux-gnu`): with `profile = "library"` and `llvm.assertions = true` in `config.toml`, running `x test library/std --stage 1` results in:

I was just able to reproduce, seems like I was running into some cache bug. Opened https://github.com/llvm/llvm-project/issues/105747, I'll just change x86 to be disabled under reliable_f16_math.

Reply:
Thanks for the thorough explanation, that makes a lot of sense. I changed it back to only `x86_64`, which won't affect anything until we change the `(_, "linux" | "macos") => true,` line.

Since you are much more aware of platform support for these types than me, feel free to put up a PR making that change if you are interested (and superseding https://github.com/rust-lang/rust/pull/129385 if you'd like, I can just close that one since it's wrong anyway).